webapp/utils/post_utils.jsx

```diff
@@ -30,3 +30,32 @@ export function getProfilePicSrcForPost(post, timestamp) {
 
     return src;
 }
+
+export function messageHistoryHandler(e) {
+    if ((e.ctrlKey || e.metaKey) && !e.altKey && !e.shiftKey && (e.keyCode === Constants.KeyCodes.UP || e.keyCode === Constants.KeyCodes.DOWN)) {
+        if (this.state.messageText !== '' && this.currentLastMsgIndex === PostStore.getHistoryLength()) {
```

Comment:
The fix for this is to create an external handing system that is shared by both center and RHS. Then pass it parameters about messages being posted and it deals with the rest. Instead of this strangeness.